trace/kafka/kafka.go

```diff
@@ -123,21 +127,33 @@ func (ap *AsyncProducer) propagateError() {
 	}
 }
 
-func createProducerMessage(ctx context.Context, msg *Message, sp opentracing.Span) (*sarama.ProducerMessage, error) {
+func createProducerMessage(ctx context.Context, msg *Message, sp opentracing.Span, enc encoding.EncodeFunc, ct string) (*sarama.ProducerMessage, error) {
```

Comment:
since enc and ct are fields of the AsyncPRoducer we could make this function have a pointer receiver on it and avoid those 2 arguments.

Reply:
Since this is not exposed in the API, I think you're right.